### PR TITLE
STB-2792 - Add MX domain validation

### PR DESCRIPTION
**Pull request by snowy-snowdon**
Added more validation in user creation. Validation now checks for MX records of the email domain during adding user details, ensuring that only email addresses with valid MX records (i.e., domains capable of receiving email) are accepted.

Screenshot 2025-09-09 at 16 14 00

**Email validation feature:**

* Added a new `EmailValidationService` that checks for MX records on the domain of a given email address, using DNS lookups to determine if the domain can receive emails.
* Integrated the `EmailValidationService` into `UserController` and updated the user creation logic to reject email addresses whose domains lack valid MX records, providing a user-friendly error message. [[1]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R68) [[2]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R97-R104) [[3]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R337-R341)

**Testing and test coverage:**

* Added `EmailValidationServiceTest` with comprehensive unit tests to cover various edge cases and ensure the service handles invalid input and DNS failures gracefully.
* Updated `UserControllerTest` to mock and inject the new `EmailValidationService` dependency. [[1]](diffhunk://#diff-21057d4cac378b8df4513dea5fe6f79c549a4ccfaf968a2440e01349f6417183R61) [[2]](diffhunk://#diff-21057d4cac378b8df4513dea5fe6f79c549a4ccfaf968a2440e01349f6417183R124-R133)

**General code cleanup:**

* Refactored and reorganized imports in `UserController.java` for clarity and maintainability. [[1]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L3-R17) [[2]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885R35-R39) [[3]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L63-L77)
* Improved formatting and readability in several method calls and parameter lists. [[1]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L146-R155) [[2]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L160-R170) [[3]](diffhunk://#diff-1c3931784714b5992966f392b56dad9140ea42ba7303dfce45c2cb8533ed3885L1809)

### CODE REVIEW

File: src/main/java/uk/gov/justice/laa/portal/landingpage/service/EmailValidationService.java [hunk 1]
```diff
@@ -1,29 +1,52 @@
...
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
 @Service
+@Slf4j
 public class EmailValidationService {
 
-    private static final Logger log = LoggerFactory.getLogger(EmailValidationService.class);
+    private static final ExecutorService executor = new ThreadPoolExecutor(10, 100, 3600,
```
[the remark this answers is not in the document]
Comment (sureshkavali-moj) [reply src/main/java/uk/gov/justice/laa/portal/landingpage/service/EmailValidationService.java:23]: The reason for having ExecutorService is to get the ability to set a timeout for email domain lookup (Adam T wanted to have this). So the code here introduced to add timeout to a synchronous call using executor service. The call generally takes 30-50 nano seconds and shouldn't be a problem. The the approach we are taking here is to have a thread pool of 10 to start with and increase it to 100 max if load demands and release the threads if no much load and keep 10 threads as base. The MX lookup call is synchronous, so we need to add some external mechanism to achieve this.

File: src/main/java/uk/gov/justice/laa/portal/landingpage/service/EmailValidationService.java [hunk 2]
```diff
@@ -1,29 +1,52 @@
...
+            return hasMxRecords(email);
+        });
+
+
+        try {
+            return future.get(30, TimeUnit.SECONDS);
+        } catch (TimeoutException timeoutEx) {
+            throw new RuntimeException("The email domain validation took longer than expected. Possibly the email domain is invalid!", timeoutEx);
+        } catch (Exception ex) {
+            throw new RuntimeException("Error while performing email domain validation. Possibly the email domain is invalid!", ex);
```
[the remark this answers is not in the document]
Comment (sureshkavali-moj) [reply src/main/java/uk/gov/justice/laa/portal/landingpage/service/EmailValidationService.java]: The idea here is to show Global error page when timeout or any other exception happens while doing email domain validation. So catching these to log the root cause and throw a generic error message to controller.

File: src/main/java/uk/gov/justice/laa/portal/landingpage/service/EmailValidationService.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (sureshkavali-moj) [reply src/main/java/uk/gov/justice/laa/portal/landingpage/service/EmailValidationService.java:23]: This may not be a problem once we onboard the external users initially.
